parsers/US_NEISO.py

```diff
@@ -225,11 +231,110 @@ def fetch_exchange(
     return exchanges.to_list()
 
 
+def fetch_wind_solar_forecasts(
+    zone_key: ZoneKey = US_NEISO_KEY,
+    session: Session | None = None,
+    target_datetime: datetime | None = None,
+    logger: Logger = getLogger(__name__),
+) -> list[dict[str, Any]]:
+    """Requests wind and solar power forecast in MW for 7 days"""
+
+    # Datetime
+    target_datetime = (
+        datetime.now(timezone.utc)
+        if target_datetime is None
+        else target_datetime.astimezone(timezone.utc)
+    )
+    target_datetime = target_datetime.astimezone(tz=ZoneInfo("America/New_York"))
+    if (
+        target_datetime.hour <= 10
+    ):  # the data is updated every day at approximately 10:00:00-05:00
+        target_datetime = target_datetime - timedelta(days=1)
+    target_datetime_string = target_datetime.strftime("%Y%m%d")
+
+    # Request url
+    session = session or Session()
+    merged_data = None
+    for data_type in ["solar", "wind"]:
+        # Get cookies by accessing the forecast pages
+        session.get(
+            "https://www.iso-ne.com/isoexpress/web/reports/operations/-/tree/seven-day-"
+            + data_type
+            + "-power-forecast"
+        )
+        response = session.get(
+            "https://www.iso-ne.com/transform/csv/"
+            + data_type[0]
+            + "phf?start="
+            + target_datetime_string
+        )
+
+        # Get dataframe
+        df = pd.read_csv(
+            io.StringIO(response.content.decode("utf8")),
+            skiprows=[0, 1, 2, 3, 4, 5],
+            skipfooter=1,
+            engine="python",
+        ).drop_duplicates()
+
+        df = df.drop(columns=["D", "Date"])
+
+        data = df.melt(
+            id_vars=["Hour Ending"],
+            var_name="Date",
+            value_name=data_type,
+        ).dropna(subset=[data_type])
+
+        if merged_data is None:
+            # For the first iteration (solar), just store the data
+            merged_data = data
+        else:
+            # For the second iteration (wind), merge with the first data
+            merged_data = pd.merge(
+                merged_data, data, on=["Hour Ending", "Date"], how="outer"
+            )
+
+    # Combine Date and Hour Ending into a datetime iso column
+    merged_data["datetime"] = pd.to_datetime(
+        merged_data["Date"]
+        + " "
+        + (merged_data["Hour Ending"] - 1).astype(str)
+        + ":00",
+        format="%m/%d/%Y %H:%M",
+    )
+    merged_data["datetime_iso"] = merged_data["datetime"].dt.strftime(
+        "%Y-%m-%dT%H:%M:%S"
+    )
+    data_final = merged_data[["datetime_iso", "solar", "wind"]]
```

Comment:
Everything looks good functionality wise but this part has a lot of type issues. If it's possible to fix that would be great otherwise we can merge it without the fixes.

Reply:
I mostly meant this:
<img width="587" alt="image" src="https://github.com/user-attachments/assets/25a16bc3-d66c-4a3c-8f2f-52178e2882a5" />

But we can merge it without fixing that.